src/test/java/utils/GeneralConfigModule.java

```diff
@@ -21,15 +21,28 @@ public void applyToConfigBuilder(SchemaGeneratorConfigBuilder builder) {
 		builder.forTypesInGeneral()
 				.withSubtypeResolver(new ClassGraphSubtypeResolver())
 				.withCustomDefinitionProvider(descriptionProvider)
-				.withTypeAttributeOverride(descriptionProvider);
+				.withTypeAttributeOverride(descriptionProvider)
+				//.withCustomDefinitionProvider(new ObjectIdToCustomSchemaProvider())
+		;
+		builder.forFields()
+				.withTargetTypeOverridesResolver(this::collectTargetTypeOverrides);
 		builder.forMethods()
+				.withTargetTypeOverridesResolver(this::collectTargetTypeOverrides)
```

Comment:
The `TargetTypeOverridesResolver` allows you to keep your hands "clean", i.e., no need to manually generate a schema. Just declare the two possible types, for which the normal schema generation logic applies.